### PR TITLE
refactor: Avoid locking tx pool cs thrice

### PR DESCRIPTION
**Pull request by maflcko**
`addUnchecked` is (outside the tests) only called by ATMP, which already takes the tx pool read lock. So locking it twice more in both `addUnchecked` methods seems redundant.

Similarly `CalculateMemPoolAncestors` is (beside once in the wallet) only called in contexts, where the tx pool lock is already taken. So remove the lock there as well.

### CODE REVIEW

**Comment by promag** [comment]
Tested ACK. Verified all OP claims.

`CalculateMemPoolAncestors` is called in:
 https://github.com/bitcoin/bitcoin/blob/ef4fac0ea5b4891f4529e4b59dfd1f7aeb3009b5/src/miner.h#L173
https://github.com/bitcoin/bitcoin/blob/ef4fac0ea5b4891f4529e4b59dfd1f7aeb3009b5/src/policy/rbf.h#L26
https://github.com/bitcoin/bitcoin/blob/ef4fac0ea5b4891f4529e4b59dfd1f7aeb3009b5/src/txmempool.h#L680
which already have the annotation.

**Comment by MarcoFalke** [comment]
Rebased

File: src/test/mempool_tests.cpp [hunk 1]
```diff
@@ -55,6 +55,7 @@ BOOST_AUTO_TEST_CASE(MempoolRemoveTest)
 
 
     CTxMemPool testPool;
+    LOCK(testPool.cs);
```
Comment (Empact) [inline src/test/mempool_tests.cpp]: nit: could move to 65 since `removeRecursive` locks separately

File: src/test/policyestimator_tests.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@ BOOST_AUTO_TEST_CASE(BlockPolicyEstimates)
 {
     CBlockPolicyEstimator feeEst;
     CTxMemPool mpool(&feeEst);
+    LOCK(mpool.cs);
```
Comment (Empact) [inline src/test/policyestimator_tests.cpp]: nit: could move to 55

**Comment by Empact** [comment]
utACK fa5ed4f

**Review by kallewoof — COMMENTED** [review]
utACK fa5ed4f8d2c4cb3507bcc2460725d483f2e5789c

**Pull request merged**